sw/device/tests/main.c

```diff
@@ -12,14 +12,15 @@
 int main(int argc, char **argv) {
   test_status_set(kTestStatusInTest);
 
+  // Run the SW test which is fully contained within `test_main()`.
```

Comment:
The reason UART was initialized before invoking `test_main` was because its is common for ALL tests for enabling logging. That way, the UART initialization code wouldn't need to be replicated in each individual tests' `test_main()` method. 

This change breaks logging in all other tests. Please see the CI results (of this PR):
https://dev.azure.com/lowrisc/opentitan/_build/results?buildId=14785&view=logs&j=924ff16e-cbb4-544c-ec55-782cc11682e2&t=00abf156-abd9-560b-8b00-bfa24bd48a48

In the `aes_test` for example, the log message "Running AES test" is missing. 

How about we re-initialize the UART with vanilla settings at the end of `dif_plic_sanitytest` and `dif_uart_sanitytest` instead?

Reply:
I think probably a hybrid between your suggestion and Sam's below is the best bet.